handlers/aws/utils.py

```diff
@@ -378,32 +370,63 @@ def get_account_id_from_arn(lambda_arn: str) -> str:
     return arn_components[4]
 
 
-def get_log_group_arn_and_region_from_log_group_name(log_group_name: str, log_stream_name: str) -> tuple[str, str]:
-    """
-    Return cloudwatch log group arn given a log group name
-    """
-    logs_client = get_cloudwatch_logs_client()
-
-    describe_log_streams_kwargs = {"logGroupName": log_group_name, "logStreamNamePrefix": log_stream_name, "limit": 50}
-    while True:
-        log_streams = logs_client.describe_log_streams(**describe_log_streams_kwargs)
-
-        assert "logStreams" in log_streams
-
-        for log_stream in log_streams["logStreams"]:
-            if "logStreamName" in log_stream and log_stream["logStreamName"] == log_stream_name:
-                log_stream_arn = log_stream["arn"]
-                region = log_stream_arn.split(":")[3]
-
-                return log_stream_arn, region
-
-        if "nextToken" in log_streams and len(log_streams["nextToken"]) > 0:
-            describe_log_streams_kwargs["nextToken"] = log_streams["nextToken"]
-        else:
-            describe_log_streams_kwargs["nextToken"] = ""
-            break
-
-    raise ValueError("Cannot find cloudwatch log stream ARN")
+def get_input_from_log_group_subscription_data(
+    config: Config, account_id: str, log_group_name: str, log_stream_name: str
+) -> tuple[str, Optional[Input]]:
+    """
+    This function is not less resilient than the previous get_log_group_arn_and_region_from_log_group_name()
+    We avoid to call the describe_log_streams on the logs' client, since we have no way to apply the proper
+    throttling because we'd need to know the number of concurrent lambda running at the time of the call.
+    The list of regions is unluckily hardcoded: there's no API to get the list for logs (the most similar API is in ec2:
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2/client/describe_regions.html - but we
+    would need to add IAM permissions for it).
+    We must keep the hardcoded list up to date or find a way to sync it automatically
+    """
+    for region in [
```

Comment:
I'd personally rather add the permissions to EC2:DescribeRegions instead of having to maintain this. We would stil have to rely on what cached in botocore, but would be easier to maintain.
Just my opinion, I wouldn't block the PR for this as they are both compromises and not optimal solutions.

Reply:
I would consider adding `EC2:DescribeRegions` for ease of maintenance. 

But I believe we should ship this fix as soon as it is ready to users having this issue.